### DISCUSSION TITLE
Help Magick++ Setup

### DISCUSSION THREAD

**Discussion by hsestagio** [ImageMagick/ImageMagick#4216] [Help]
Hi, im working on a windows c++ project and i need to use ImageMagick in it. I already included the "lib" and "include" folders and all lib files in project properties. When i include "Magick++.h" in project, after i compile, it shows 500+ errors C2144, C4430, E0020, etc...
Am i missing something?

**Comment by snibgo** [comment]
What is your build environment? GNU toolset, or Microsoft Visual C, or what? With the GNU tools, my basic build command for Magick++ programs is:
```
c++ -o "$1".exe "$1".cpp -Wall -Wa,-march=corei7,-mtune=corei7 `pkg-config --cflags --libs Magick++`
```
I don't use other environments, and can't advise on them.

You IM distribution contains sample C++ programs, such as button.cpp. I suggest you use these to test your development environment.
